fedimint-api-client/src/api/global_federation_api_with_cache.rs

```diff
@@ -430,4 +431,146 @@ where
             })
         }
     }
+
+    async fn api_announcements(
+        &self,
+        guardian_pub_keys: &BTreeMap<PeerId, PublicKey>,
+    ) -> FederationResult<BTreeMap<PeerId, SignedApiAnnouncement>> {
+        let guardian_pub_keys = guardian_pub_keys.clone();
+        let peer_responses = self
+            .request_with_strategy(
+                FilterMapThreshold::new(
+                    move |peer_id, announcements: Vec<SignedApiAnnouncement>| {
+                        // If any of the announcements is invalid something is fishy with that
+                        // guardian and we ignore all its responses
+                        for announcement in &announcements {
+                            let Some(guardian_pub_key) = guardian_pub_keys.get(&peer_id) else {
```

Comment:
> This seems to be the public key of the peer that returned the announcements, not the one that signed it originally?

Now that your other comment is fixed we need to bail here because we are actually using the peer id of the announcer, which is untrusted input.